openedx/features/subscriptions/api/v1/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+HTTP endpoints for interacting with subscriptions.
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import ListAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from django.http import Http404
+
+from lms.djangoapps.commerce.api.v1.permissions import ApiKeyOrModelPermission
+from openedx.core.lib.api.mixins import PutAsCreateMixin
+from openedx.features.subscriptions.models import UserSubscription
+from openedx.features.subscriptions.api.v1.serializers import UserSubscriptionSerializer
+
+
+class SubscriptionsListView(ListAPIView):
+    """
+    List subscriptions.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserSubscriptionSerializer
+
+    def get_queryset(self):
+        valid_subscriptions = True if self.request.query_params.get('valid') == 'true' else False
+        user_id = self.request.query_params.get('user', self.request.user.id)
+        if valid_subscriptions:
+            return UserSubscription.get_valid_subscription(user_id=user_id)
+
+        return UserSubscription.objects.filter(user_id=self.request.user.id)
+
+
+class SubscriptionRetrieveUpdateView(PutAsCreateMixin, RetrieveUpdateAPIView):
+    """
+    Retrieve, update, or create UserSubscriptions.
+    """
+    lookup_field = 'subscription_id'
+    lookup_url_kwarg = 'subscription_id'
+    model = UserSubscription
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (ApiKeyOrModelPermission,)
+    queryset = UserSubscription.objects.all()
+    serializer_class = UserSubscriptionSerializer
+
+    def get_object(self, queryset=None):
+        user_subscription = UserSubscription.objects.filter(**self.kwargs).first()
```

Comment:
Why first? A user can have multiple subscriptions.

Reply:
This is a retrieve/update/create view. Also, kwargs have `lookup_field` as key which is `subscription_id`. There can only be one subscription against a subscription id for a user at max. but I need to filter it on user as well. going to make that change